ca/ca.go

```diff
@@ -414,6 +415,7 @@ func (ca *certificateAuthorityImpl) IssueCertificateForPrecertificate(ctx contex
 
 	_, issuanceToken, err := issuer.Prepare(certProfile, issuanceReq)
 	if err != nil {
+		ca.certProfileCount.With(prometheus.Labels{"purpose": string(certType), "profile": req.CertProfile.Name, "hash": hex.EncodeToString(req.CertProfile.Hash), "status": "preparation failure"}).Inc()
```

Comment:
This is a fairly long line and it's repeated several times. You could simplify by using [CurryWith](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#CounterVec.CurryWith) early in the function for the things that always stay the same (profile and hash). Then at each increment site you would only need to set the labels that are actually distinct (purpose and status).

Reply:
Interesting, I'll have to play with that. That's a new concept to me.

Reply:
I really like that. Good suggestion!